code/modules/mob/living/carbon/resist.dm

```diff
@@ -1,22 +1,24 @@
-
 /mob/living/carbon/process_resist()
 
 	//drop && roll
 	if(on_fire && !buckled)
-		fire_stacks -= 1.2
+		var/obj/effect/decal/cleanable/foam/extinguisher_foam = locate() in src.loc

```

Comment:
I suspect this is bad. Shouldn't we make extinguisher foam itself process() and apply its effects?

Reply:
Arrow was against foam passively extinguish.

Reply:
I'm trying to decide which is a better approach in terms of OOP. I guess it doesn't matter in the end, though.